src/Facade/Services/WorkstationFacadeService.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace Linn.Stores2.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Stores2.Domain.LinnApps;
+    using Linn.Stores2.Domain.LinnApps.Exceptions;
+    using Linn.Stores2.Domain.LinnApps.Stores;
+    using Linn.Stores2.Facade.Common;
+    using Linn.Stores2.Resources.Stores;
+
+    public class WorkstationFacadeService : AsyncFacadeService<Workstation, string, WorkstationResource, WorkstationResource, WorkstationSearchResource>
+    {
+        private readonly IRepository<Workstation, string> repository;
+
+        private readonly IRepository<Employee, int> employeeRepository;
+
+        private readonly IRepository<Cit, string> citRepository;
+
+        public WorkstationFacadeService(
+            IRepository<Workstation, string> repository,
+            IRepository<Employee, int> employeeRepository,
+            IRepository<Cit, string> citRepository,
+            ITransactionManager transactionManager,
+            IBuilder<Workstation> resourceBuilder)
+            : base(repository, transactionManager, resourceBuilder)
+        {
+            this.repository = repository;
+            this.employeeRepository = employeeRepository;
+            this.citRepository = citRepository;
+        }
+
+        protected override async Task<Workstation> CreateFromResourceAsync(
+            WorkstationResource resource,
+            IEnumerable<string> privileges = null)
+        {
+            var workstation = await this.repository.FindByIdAsync(resource.WorkstationCode);
+
+            if (workstation != null)
+            {
+                throw new WorkstationException("Work station already exists!");
+            }
+
```

Comment:
we don't tend to do this sort of thing, mainly because the database wont let you add duplicate PKs anyway. I'm not saying its a bad idea - its actually a probably nicer way to handle this sort of scenario - but I'm not sure its code I would have in my facade service